addon/editor/input-handlers/paste-handler.ts

```diff
@@ -0,0 +1,81 @@
+import PernetRawEditor from "@lblod/ember-rdfa-editor/utils/ce/pernet-raw-editor";
+import {PropertyState} from "@lblod/ember-rdfa-editor/model/util/types";
+import ModelRangeUtils from "@lblod/ember-rdfa-editor/model/util/model-range-utils";
+import HTMLInputParser, {LIMITED_SAFE_TAGS} from "@lblod/ember-rdfa-editor/utils/html-input-parser";
+import {MisbehavedSelectionError} from "@lblod/ember-rdfa-editor/utils/errors";
+import {taskFor} from "ember-concurrency-ts";
+import {InputHandler} from "@lblod/ember-rdfa-editor/editor/input-handlers/input-handler";
+import {HandlerResponse} from "@lblod/ember-rdfa-editor/editor/input-handlers/handler-response";
+
+export default class PasteHandler extends InputHandler {
+  constructor({rawEditor}: {rawEditor: PernetRawEditor}) {
+    super(rawEditor);
+  }
+
+  // see https://www.w3.org/TR/clipboard-apis/#paste-action for more info
+  handleEvent(event: ClipboardEvent, pasteHTML: boolean, pasteExtendedHTML: boolean): HandlerResponse {
+    const clipboardData = event.clipboardData;
+
+    if (!clipboardData) {
+      //TODO: if no clipboardData found, do we want an error?
```

Comment:
If you're not sure, maybe add a warning?
best to use a logger, imported from logging-utils
```ts
const logger = createLogger("PasteHandler");  // you can also make this a static property
logger.warn("No clipboardData object found, ignoring paste");
````